src/utils/requests.js

```diff
@@ -0,0 +1,51 @@
+import { useCallback } from "react";
+
+const base_url="https://localhost:8000";
+
+const sendRequest = useCallback(
+  async (url, method = "GET", body = null, headers = {}) => {
```

Comment:
It's better to let `method` and body be a necessary parameter - these are essential parts of every request and it's better if each request explicitly mentions the method and body.